embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/comms/delivery/EmbracePendingApiCallsSender.kt

```diff
@@ -84,16 +95,15 @@
     /**
      * Returns true if there is an active pending retry task
      */
-    fun isDeliveryTaskActive(): Boolean =
-        lastDeliveryTask?.let { task ->
-            !task.isCancelled && !task.isDone
-        } ?: false
+    fun isDeliveryTaskActive(): Boolean = lastDeliveryTask?.let { task ->
+        !task.isCancelled && !task.isDone
+    } ?: false
 
     /**
      * Return true if the conditions are met for a delivery task to be scheduled
      */
     private fun shouldScheduleDelivery(): Boolean {
-        return !isDeliveryTaskActive() && pendingApiCalls.hasAnyPendingApiCall()
+        return !isDeliveryTaskActive() && pendingApiCalls.hasPendingApiCallsToSend()
```

Comment:
I'm wondering if you can check to see if this endpoint to see if it's rate limited instead of asking `pendingApiCalls` to do so in `hasPendingApiCallsToSend()` so that it won't have to know about rate limiting?